### ISSUE TITLE
Hive plugin doesn't work

### ISSUE THREAD

**Issue by tracycarley** [aws/aws-mwaa-local-runner#4]
I follow the [tutorial](https://docs.aws.amazon.com/mwaa/latest/userguide/samples-hive.html) for custom plugins for Hive and Hadoop. Unfortunately, it doesn't work. I was able to reproduce the issue both locally and on the MWAA. The issue indicates the Python version is wrong, so with the thirft version.

Can we update the tutorial with some testing point?
```
local-runner_1  | Installing requirements.txt
local-runner_1  | You must give at least one requirement to install (see "pip help install")
local-runner_1  | [2021-05-05 16:30:01,884] {{plugins_manager.py:166}} ERROR - unsupported operand type(s) for +: 'NoneType' and 'str'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/hive_plugin.py", line 8, in <module>
local-runner_1  |     os.environ["CLASSPATH"] = os.getenv("CLASSPATH") + ":/usr/local/airflow/plugins/apache-hive-3.1.2-bin/lib"
local-runner_1  | TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
local-runner_1  | [2021-05-05 16:30:01,886] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/hive_plugin.py
local-runner_1  | [2021-05-05 16:30:02,074] {{plugins_manager.py:166}} ERROR - Missing parentheses in call to 'print'. Did you mean print("Would run:")? (hcat.py, line 143)
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 724, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 860, in get_code
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/hcatalog/bin/hcat.py", line 143
local-runner_1  |     print "Would run:"
local-runner_1  |                      ^
local-runner_1  | SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Would run:")?
local-runner_1  | [2021-05-05 16:30:02,074] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/hcatalog/bin/hcat.py
local-runner_1  | [2021-05-05 16:30:02,093] {{plugins_manager.py:166}} ERROR - invalid syntax (hcat_server.py, line 29)
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 724, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 860, in get_code
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/hcatalog/sbin/hcat_server.py", line 29
local-runner_1  |     print "Usage: %s [--config confdir] COMMAND" % (sys.argv[0])
local-runner_1  |                                                ^
local-runner_1  | SyntaxError: invalid syntax
local-runner_1  | [2021-05-05 16:30:02,093] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/hcatalog/sbin/hcat_server.py
local-runner_1  | [2021-05-05 16:30:02,297] {{plugins_manager.py:166}} ERROR - No module named 'FacebookService'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303/FacebookBase.py", line 23, in <module>
local-runner_1  |     import FacebookService
local-runner_1  | ModuleNotFoundError: No module named 'FacebookService'
local-runner_1  | [2021-05-05 16:30:02,298] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303/FacebookBase.py
local-runner_1  | [2021-05-05 16:30:02,316] {{plugins_manager.py:166}} ERROR - No module named 'ttypes'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303/FacebookService.py", line 8, in <module>
local-runner_1  |     from ttypes import *
local-runner_1  | ModuleNotFoundError: No module named 'ttypes'
local-runner_1  | [2021-05-05 16:30:02,318] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303/FacebookService.py
local-runner_1  | [2021-05-05 16:30:02,333] {{plugins_manager.py:166}} ERROR - No module named 'ttypes'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303/constants.py", line 8, in <module>
local-runner_1  |     from ttypes import *
local-runner_1  | ModuleNotFoundError: No module named 'ttypes'
local-runner_1  | [2021-05-05 16:30:02,335] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303/constants.py
local-runner_1  | [2021-05-05 16:30:02,360] {{plugins_manager.py:166}} ERROR - Missing parentheses in call to 'print'. Did you mean print(msg)? (fb303_simple_mgmt.py, line 58)
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 724, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 860, in get_code
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303_scripts/fb303_simple_mgmt.py", line 58
local-runner_1  |     print msg
local-runner_1  |             ^
local-runner_1  | SyntaxError: Missing parentheses in call to 'print'. Did you mean print(msg)?
local-runner_1  | [2021-05-05 16:30:02,360] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/fb303_scripts/fb303_simple_mgmt.py
local-runner_1  | [2021-05-05 16:30:02,377] {{plugins_manager.py:166}} ERROR - No module named 'ttypes'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/hive_serde/constants.py", line 10, in <module>
local-runner_1  |     from ttypes import *
local-runner_1  | ModuleNotFoundError: No module named 'ttypes'
local-runner_1  | [2021-05-05 16:30:02,379] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/hive_serde/constants.py
local-runner_1  | [2021-05-05 16:30:02,403] {{plugins_manager.py:166}} ERROR - No module named 'ttypes'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/queryplan/constants.py", line 10, in <module>
local-runner_1  |     from ttypes import *
local-runner_1  | ModuleNotFoundError: No module named 'ttypes'
local-runner_1  | [2021-05-05 16:30:02,404] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/queryplan/constants.py
local-runner_1  | [2021-05-05 16:30:02,436] {{plugins_manager.py:166}} ERROR - No module named 'SCons'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/TSCons.py", line 21, in <module>
local-runner_1  |     from SCons.Builder import Builder
local-runner_1  | ModuleNotFoundError: No module named 'SCons'
local-runner_1  | [2021-05-05 16:30:02,438] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/TSCons.py
local-runner_1  | [2021-05-05 16:30:02,470] {{plugins_manager.py:166}} ERROR - No module named 'TProtocol'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/protocol/TBinaryProtocol.py", line 20, in <module>
local-runner_1  |     from TProtocol import *
local-runner_1  | ModuleNotFoundError: No module named 'TProtocol'
local-runner_1  | [2021-05-05 16:30:02,471] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/protocol/TBinaryProtocol.py
local-runner_1  | [2021-05-05 16:30:02,515] {{plugins_manager.py:166}} ERROR - No module named 'ttypes'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/reflection/limited/constants.py", line 8, in <module>
local-runner_1  |     from ttypes import *
local-runner_1  | ModuleNotFoundError: No module named 'ttypes'
local-runner_1  | [2021-05-05 16:30:02,517] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/reflection/limited/constants.py
local-runner_1  | [2021-05-05 16:30:02,536] {{plugins_manager.py:166}} ERROR - No module named 'BaseHTTPServer'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/server/THttpServer.py", line 20, in <module>
local-runner_1  |     import BaseHTTPServer
local-runner_1  | ModuleNotFoundError: No module named 'BaseHTTPServer'
local-runner_1  | [2021-05-05 16:30:02,537] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/server/THttpServer.py
local-runner_1  | [2021-05-05 16:30:02,545] {{plugins_manager.py:166}} ERROR - No module named 'Queue'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/server/TNonblockingServer.py", line 28, in <module>
local-runner_1  |     import Queue
local-runner_1  | ModuleNotFoundError: No module named 'Queue'
local-runner_1  | [2021-05-05 16:30:02,547] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/server/TNonblockingServer.py
local-runner_1  | [2021-05-05 16:30:02,549] {{plugins_manager.py:166}} ERROR - invalid syntax (TServer.py, line 84)
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 724, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 860, in get_code
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/server/TServer.py", line 84
local-runner_1  |     except TTransport.TTransportException, tx:
local-runner_1  |                                          ^
local-runner_1  | SyntaxError: invalid syntax
local-runner_1  | [2021-05-05 16:30:02,550] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/server/TServer.py
local-runner_1  | [2021-05-05 16:30:02,568] {{plugins_manager.py:166}} ERROR - No module named 'TTransport'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/THttpClient.py", line 20, in <module>
local-runner_1  |     from TTransport import *
local-runner_1  | ModuleNotFoundError: No module named 'TTransport'
local-runner_1  | [2021-05-05 16:30:02,569] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/THttpClient.py
local-runner_1  | [2021-05-05 16:30:02,572] {{plugins_manager.py:166}} ERROR - invalid syntax (TSocket.py, line 78)
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 724, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 860, in get_code
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TSocket.py", line 78
local-runner_1  |     except socket.error, e:
local-runner_1  |                        ^
local-runner_1  | SyntaxError: invalid syntax
local-runner_1  | [2021-05-05 16:30:02,573] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TSocket.py
local-runner_1  | [2021-05-05 16:30:02,584] {{plugins_manager.py:166}} ERROR - No module named 'cStringIO'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TTransport.py", line 20, in <module>
local-runner_1  |     from cStringIO import StringIO
local-runner_1  | ModuleNotFoundError: No module named 'cStringIO'
local-runner_1  | [2021-05-05 16:30:02,585] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TTransport.py
local-runner_1  | [2021-05-05 16:30:02,595] {{plugins_manager.py:166}} ERROR - No module named 'zope.interface'
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TTwisted.py", line 19, in <module>
local-runner_1  |     from zope.interface import implements, Interface, Attribute
local-runner_1  | ModuleNotFoundError: No module named 'zope.interface'
local-runner_1  | [2021-05-05 16:30:02,596] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TTwisted.py
local-runner_1  | [2021-05-05 16:30:02,641] {{plugins_manager.py:166}} ERROR - Missing parentheses in call to 'print'. Did you mean print("Cannot determine the container size")? (package.py, line 30)
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 724, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 860, in get_code
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/scripts/llap/yarn/package.py", line 30
local-runner_1  |     print "Cannot determine the container size"
local-runner_1  |                                               ^
local-runner_1  | SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Cannot determine the container size")?
local-runner_1  | [2021-05-05 16:30:02,642] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/scripts/llap/yarn/package.py
```

**Comment by tracycarley** [comment]
> try changing the following line in hive_plugin.py:
> 
> `os.environ["CLASSPATH"] = os.getenv("CLASSPATH") + ":/usr/local/airflow/plugins/apache-hive-3.1.2-bin/lib" `
> 
> to
> 
> `os.environ["CLASSPATH"] = "/usr/local/airflow/dags/classpath/*:/usr/local/airflow/plugins/apache-hive-3.1.2-bin/lib" `

Tried the solution but still seeing issues - implying Python2 code

```
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TTwisted.py", line 19, in <module>
local-runner_1  |     from zope.interface import implements, Interface, Attribute
local-runner_1  | ModuleNotFoundError: No module named 'zope.interface'
local-runner_1  | [2021-05-11 15:50:57,182] {{plugins_manager.py:167}} ERROR - Failed to import plugin /usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/lib/py/thrift/transport/TTwisted.py
local-runner_1  | [2021-05-11 15:50:57,192] {{plugins_manager.py:166}} ERROR - Missing parentheses in call to 'print'. Did you mean print("Cannot determine the container size")? (package.py, line 30)
local-runner_1  | Traceback (most recent call last):
local-runner_1  |   File "/usr/local/lib/python3.7/site-packages/airflow/plugins_manager.py", line 160, in <module>
local-runner_1  |     m = imp.load_source(namespace, filepath)
local-runner_1  |   File "/usr/lib64/python3.7/imp.py", line 171, in load_source
local-runner_1  |     module = _load(spec)
local-runner_1  |   File "<frozen importlib._bootstrap>", line 696, in _load
local-runner_1  |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 724, in exec_module
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 860, in get_code
local-runner_1  |   File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
local-runner_1  |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
local-runner_1  |   File "/usr/local/airflow/plugins/hive_plugin/apache-hive-3.1.2-bin/scripts/llap/yarn/package.py", line 30
local-runner_1  |     print "Cannot determine the container size"
```

**Comment by john-jac** [comment]
Python 2.7 is included in the 3.7.9 version that MWAA deploys.  It may not be in local-runner.

**Comment by tracycarley** [comment]
> Python 2.7 is included in the 3.7.9 version that MWAA deploys. It may not be in local-runner.

Thanks. That won't work with MWAA prod env right? Given Python3 is the only supported version.

**Comment by john-jac** [comment]
> > Python 2.7 is included in the 3.7.9 version that MWAA deploys. It may not be in local-runner.
> 
> Thanks. That won't work with MWAA prod env right? Given Python3 is the only supported version.

Python 2.7 exists on MWAA.  You can try it yourself by running a BashOperator with command "python --version"

**Comment by tracycarley** [comment]
> Python 2.7 is included in the 3.7.9 version that MWAA deploys. It may not be in local-runner.

Do we have any instructions on how to include this in the local-runner? through certain config file?

**Comment by john-jac** [comment]
Please use airflow-ignore per https://docs.aws.amazon.com/mwaa/latest/userguide/samples-hive.html
